hack/generated/pkg/genruntime/base_types.go

```diff
@@ -37,3 +202,43 @@ type ArmResourceSpec interface {
 
 	GetName() string
 }
+
+// ArmResourceStatus is an ARM resource status
+type ArmResourceStatus interface {
+	// GetId() string
+}
+
+type ArmResource interface {
```

Comment:
It needs to be in `genruntime` because it's actually used by all of the code generated code (in asserts ensuring they impl the interface). `genruntime` is the shared lib that contains all of the "helpers" used by the generated code, and some common interface definitions used by both the generated code and the generic controller.

Reply:
Ok, makes sense.